### DISCUSSION TITLE
How do I display the app while also calling another function simultaneously?

### DISCUSSION THREAD

**Discussion by TechPro424** [lawsie/guizero#512] [Q&A] [answered]
I'm trying to recreate the UNO card game using GuiZero, and all my backend logic is in a file called logic, and runs when I call the function `logic.main()`
However, on importing logic in my entrypoint file (the file I run to start the program), and calling `logic.main()` after `app.display()`, `logic.main()` is not called as `app.display()` is blocking further execution.
`logic.main()` is a function that runs in an infinite loop using `while True:` till one of the players has no cards, so I cannot call that before displaying the app.
How do I display the app while also calling the `logic.main()` function?

**Comment by TechPro424** [comment]
I tried using `app.after(10, logic.main)`.
The function was called, but the app stopped responding.

**Reply by bsimmo** [reply]
> I tried using `app.after(10, logic.main)`. The function was called, but the app stopped responding.

With the way you are using it, you need to remove your while loop in the game code.
The app.after calls the 'what was the while loop' and runs through that bit of code.

If you need any further help, pop to the RPi forum and ask in the python section.

(While this isn't Pi specific, it is/was developed from within the RPi Foundation, even if they don't work there any more)

**Reply by bsimmo** [reply]
Oh ,just noticed there is now discussions z so ignore the above.
Thanks @lawsie for adding this :-)

**Comment by martinohanlon** [comment]
There is some advice here - https://lawsie.github.io/guizero/blocking/

You could also look at methods for threading and sharing state between your
GUI and your game.

However... you may find that it is simpler and easier to rewrite your logic
code to work using an event driven model.

**Reply by TechPro424** [reply]
Any examples of event driven models you can provide? I'm not too familiar with them.

**Comment by bsimmo** [comment]
Look at the sensehat example I wrote, but somehow doesn't have me as the
author.

(If on a pi you can change to sense_emu to actually run it)

There is also auto_update...  , after_repeat... , app_events...

**Reply by lawsie** [reply]
Sorry that there isn't a credit, is it this one? Very happy to add a credit in the comments (or you can add a PR for it)?
https://github.com/lawsie/guizero/blob/master/examples/sensehat_read_data.py

**Comment by martinohanlon** [comment, marked as answer]
There are lots of examples in the repo. Including a game called floodit
which give you some inspiration.

https://github.com/lawsie/guizero/blob/master/examples/floodit.py

**Reply by TechPro424** [reply]
Yes, I think I understand how to go about it now. Thank you so much!

**Comment by bsimmo** [comment]
I remember when these where edited for a tidy up by the now author, that
and many others lost the original creator. I was surprised GitHub could
loose them
